src/operations/map/MapOperations.tsx

```diff
@@ -53,20 +49,43 @@ export class MapOperations {
 
       if (selectedMapType === GlobalInsight.FOOD) {
         const newCountryIso3Data = await countryRepository.getCountryIso3Data(selectedCountryData.properties.iso3);
-        setCountryIso3Data(newCountryIso3Data);
+        if (Array.isArray(newCountryIso3Data) && newCountryIso3Data[1] === 404) {
+          setIsDataAvailable(false);
+        } else {
+          setCountryIso3Data(newCountryIso3Data);
+          setIsDataAvailable(true);
+        }
       }
 
       if (selectedMapType === GlobalInsight.NUTRITION) {
         const newRegionNutritionData = await countryRepository.getRegionNutritionData(
           selectedCountryData.properties.adm0_id
         );
+        const hasNutrition = newRegionNutritionData.features.some(
+          (feature) =>
+            feature.properties?.nutrition &&
+            typeof feature.properties.nutrition === 'object' &&
+            Object.keys(feature.properties.nutrition).length > 0
+        );
+        setIsDataAvailable(hasNutrition);
         if (newRegionNutritionData && newRegionNutritionData.features) {
           setRegionNutritionData({
             type: 'FeatureCollection',
             features: newRegionNutritionData.features as GeoJsonFeature<Geometry, GeoJsonProperties>[],
           });
         }
       }
+      if (selectedMapType === GlobalInsight.FOOD) {
+        const newRegionData = await countryRepository.getRegionData(selectedCountryData.properties.adm0_id);
+        const hasFcs = newRegionData.features.some((feature) => feature.properties?.fcs !== undefined);
+        setIsDataAvailable(hasFcs);
+        if (newRegionData && newRegionData.features) {
+          setRegionData({
+            type: 'FeatureCollection',
+            features: newRegionData.features as GeoJsonFeature<Geometry, GeoJsonProperties>[],
+          });
+        }
+      }
```

Comment:
Why did you move this to the end? The adm1data should be the first request which should be triggered when clicked on a country. It is also like this on the current new HungerMap. Please put it back to the top and also consider only this endpoint for the setIsDataAvailable. Iso3 and countryData is not relevant, just the adm1data

Reply:
i moved to the end because even if we don't have the data of country in fcs view, i always get data for the region and which make setIsDataAvailable(true). i need to move it to the end. for example ethopia, ethopia is in nutrition and ipc view but not in fcs view. but if the loop stays at the top and you make switch from ipc/nutrition view you always get true. Which shouldn't be the case